food-truck-api/src/main/java/food/truck/api/truck/TruckService.java

```diff
@@ -95,8 +91,11 @@ public Truck updateTruck(
                     .filter(tag -> !tag.isEmpty())
                     .collect(Collectors.toSet());
             truck.setTags(tags2);
+        } else {
+            truck.setTags(null);
```

Comment:
I'll need to check whether I assumed non-NULL truck tags while performing the semantic analysis stuff. There's a chance I did.